src/sage/categories/pushout.py

```diff
@@ -3472,11 +3476,11 @@ def merge(self, other):
         # integers to encode degrees of extensions.
         from sage.rings.integer import Integer
         kwds_self = dict(self.kwds.items())
-        if 'impl' in kwds_self:
-            del kwds_self['impl']
+        if 'implementation' in kwds_self:
+            del kwds_self['implementation']
         kwds_other = dict(other.kwds.items())
-        if 'impl' in kwds_other:
-            del kwds_other['impl']
+        if 'implementation' in kwds_other:
+            del kwds_other['implementation']
```

Comment:
Because `implementation` keyword argument is now already removed from `kwds` when `AlgebraicExtensionFunctor` is created, I don't think we need this part. I checked this with 
```python
        from sage.rings.integer import Integer
        kwds_self = dict(self.kwds.items())
        assert 'implementation' not in kwds_self
        kwds_other = dict(other.kwds.items())
        assert 'implementation' not in kwds_other   
```

Reply:
Removed.

Reply:
Do we still need L3553,3554, which  deletes `implementation` from `kwds_self`?